File: app/src/main/java/com/kotoba/app/data/model/Chapter.java
package com.kotoba.app.data.model;

import java.io.Serializable;

public class Chapter implements Serializable {
    private final int babNumber;
    private final String titleJa;
    private final String titleId;
    private final String theme;
    private final String level;
    private final int vocabCount;
    private final int grammarCount;

    public Chapter(int babNumber, String titleJa, String titleId, String theme, String level, int vocabCount, int grammarCount) {
        this.babNumber = babNumber;
        this.titleJa = titleJa;
        this.titleId = titleId;
        this.theme = theme;
        this.level = level;
        this.vocabCount = vocabCount;
        this.grammarCount = grammarCount;
    }

    public int getBabNumber() { return babNumber; }
    public String getTitleJa() { return titleJa; }
    public String getTitleId() { return titleId; }
    public String getTheme() { return theme; }
    public String getLevel() { return level; }
    public int getVocabCount() { return vocabCount; }
    public int getGrammarCount() { return grammarCount; }

    public String getDisplayLabel() {
        return "Bab " + babNumber + ": " + titleJa;
    }
}

File: ios/Kotoba/Models/Chapter.swift
import Foundation

public struct Chapter: Identifiable, Hashable, Codable {
    public var id: Int { babNumber }
    public let babNumber: Int
    public let titleJa: String
    public let titleId: String
    public let theme: String
    public let level: String
    public let vocabCount: Int
    public let grammarCount: Int

    public init(
        babNumber: Int,
        titleJa: String,
        titleId: String,
        theme: String,
        level: String = "N5",
        vocabCount: Int = 0,
        grammarCount: Int = 0
    ) {
        self.babNumber = babNumber
        self.titleJa = titleJa
        self.titleId = titleId
        self.theme = theme
        self.level = level
        self.vocabCount = vocabCount
        self.grammarCount = grammarCount
    }

    public var displayLabel: String {
        return "Bab \(babNumber): \(titleJa)"
    }
}
